llvm/test/tools/sycl-post-link/device-requirements/reqd-work-group-size.ll

```diff
@@ -0,0 +1,213 @@
+; Original code:
+; #include <sycl/sycl.hpp>
+
+; int main() {
+;   sycl::queue q;
+;   q.submit([&](sycl::handler &h) {
+;       h.parallel_for<class KernelA>(
+;           sycl::range<1>(32),
+;           [=](sycl::item<1> it) [[sycl::reqd_work_group_size(32)]] {});
+;     });
+;   q.submit([&](sycl::handler &h) {
+;       h.parallel_for<class KernelB>(
+;           sycl::range<1>(32),
+;           [=](sycl::item<1> it) [[sycl::reqd_work_group_size(64)]] {});
+;     });
+;   q.submit([&](sycl::handler &h) {
+;       h.parallel_for<class KernelC>(
+;           sycl::range<1>(32),
+;           [=](sycl::item<1> it) [[sycl::reqd_work_group_size(32)]] {});
+;     });
+;   return 0;
+; }
+
+; RUN: sycl-post-link -split=auto %s -o %t.table
+; RUN: FileCheck %s -input-file=%t_0.prop --check-prefix CHECK-PROP-AUTO-SPLIT-0
+; RUN: FileCheck %s -input-file=%t_1.prop --check-prefix CHECK-PROP-AUTO-SPLIT-1
+
+; CHECK-PROP-AUTO-SPLIT-0: [SYCL/device requirements]
+; CHECK-PROP-AUTO-SPLIT-0-NEXT: aspects=2|AAAAAAAAAAA
+; CHECK-PROP-AUTO-SPLIT-0-NEXT: reqd_work_group_size=2|gAAAAAAAAAAQAAAA
+
+; CHECK-PROP-AUTO-SPLIT-1: [SYCL/device requirements]
+; CHECK-PROP-AUTO-SPLIT-1-NEXT: aspects=2|AAAAAAAAAAA
+; CHECK-PROP-AUTO-SPLIT-1-NEXT: reqd_work_group_size=2|gAAAAAAAAAAIAAAA
+
+; ModuleID = '/tmp/source-3b9f8c.bc'
+
+source_filename = "llvm-link"
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+$_ZTSN4sycl3_V16detail19__pf_kernel_wrapperIZZ4mainENKUlRNS0_7handlerEE_clES4_E7KernelAEE = comdat any
+
+$_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_E7KernelA = comdat any
+
+$_ZTSN4sycl3_V16detail19__pf_kernel_wrapperIZZ4mainENKUlRNS0_7handlerEE0_clES4_E7KernelBEE = comdat any
+
+$_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE0_clES2_E7KernelB = comdat any
+
+$_ZTSN4sycl3_V16detail19__pf_kernel_wrapperIZZ4mainENKUlRNS0_7handlerEE1_clES4_E7KernelCEE = comdat any
+
+$_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE1_clES2_E7KernelC = comdat any
+
+@__spirv_BuiltInGlobalInvocationId = external dso_local local_unnamed_addr addrspace(1) constant <3 x i64>, align 32
+
+; Function Attrs: norecurse
+define weak_odr dso_local spir_kernel void @_ZTSN4sycl3_V16detail19__pf_kernel_wrapperIZZ4mainENKUlRNS0_7handlerEE_clES4_E7KernelAEE() local_unnamed_addr #0 comdat !srcloc !46 !kernel_arg_buffer_location !47 !reqd_work_group_size !48 !sycl_fixed_targets !49 !sycl_kernel_omit_args !50 {
+entry:
+  call void @__itt_offload_wi_start_wrapper()
+  %0 = load <3 x i64>, <3 x i64> addrspace(1)* @__spirv_BuiltInGlobalInvocationId, align 32, !noalias !51
+  %1 = extractelement <3 x i64> %0, i64 0
+  %cmp.i.i = icmp ult i64 %1, 2147483648
+  tail call void @llvm.assume(i1 %cmp.i.i)
+  call void @__itt_offload_wi_finish_wrapper()
+  ret void
+}
+
+declare void @__itt_offload_wi_start_wrapper()
```

Comment:
You should use `-fno-sycl-instrument-device-code` to simplify this IR: we are not interested in ITT instrumentation for the purpose of this test.